lib/auth/hawk.js

```diff
@@ -0,0 +1,119 @@
+const sdk = require('postman-collection'),
+  {
+    randomString,
+    hmac,
+    getTimeStamp
+  } = require('../helpers');
+
+/**
+ * Adds necessary headers for Hawk Authorization
+ * Uses a hashed string called MAC (Message Authentication Code) to authorize requests
+ * Reference: https://github.com/outmoded/hawk/
+ *
+ * @param {sdk.Request} request - Request Instance
+ * @param {sdk.RequestAuth} auth - Auth that needs to be applied to the given request
+ * @returns {sdk.Request} - Requst with applied auth
+ */
+function hawk (request, auth) {
+
+  // return same request if auth headers are already added
+  if (request.headers.has('authorization')) {
+    return request;
+  }
+
+  let config = {},
+    authId, authKey, app, dlg, ext, algorithm,
+    HAWK_HEADER = 'hawk.1.header',
+    nonce, hawkNormalizedString,
+    uri, host, method, port, hash,
+    mac, hawkHeaderValue, ts;
+  // creating congif obj for easy access of elements
+  auth.hawk.each((element) => {
+    config[element.key] = element.value;
+  });
+
+  // if Authid and Authkey is not provided return without change
+  if (!config.authId || !config.authKey) {
+    return request;
+  }
+
+  // Hawk Authorization required elements
+  authId = config.authId;
+  authKey = config.authKey;
+  algorithm = config.algorithm;
+
+
+  // Hawk Authorization optional elements
+  ts = config.timestamp || getTimeStamp().toString();
```

Comment:
If we are getting timestamp during the SDK generation, will this be useful when the user actually tries to use this sdk?

Reply:
>Unlike the HTTP Digest authentication scheme, Hawk provides limited protection against replay attacks which does not require prior interaction with the server. Instead, the client provides a timestamp which the server can use to prevent replay attacks outside a narrow time window. 

https://alexbilbie.com/2012/11/hawk-a-new-http-authentication-scheme/

Reply:
> If we are getting timestamp during the SDK generation, will this be useful when the user actually tries to use this sdk? 
> 
> 

Yes I did come across this question, but since hawk has option to set ts via postman app this would be helpful in that case.